platforms/core-runtime/logging/src/main/java/org/gradle/internal/deprecation/DeprecationLogger.java

```diff
@@ -67,8 +74,8 @@ protected Integer initialValue() {
 
     private static final LoggingDeprecatedFeatureHandler DEPRECATED_FEATURE_HANDLER = new LoggingDeprecatedFeatureHandler();
 
-    public synchronized static void init(ProblemDiagnosticsFactory problemDiagnosticsFactory, WarningMode warningMode, BuildOperationProgressEventEmitter buildOperationProgressEventEmitter, Problems problemsService) {
-        DEPRECATED_FEATURE_HANDLER.init(problemDiagnosticsFactory, warningMode, buildOperationProgressEventEmitter, problemsService);
+    public synchronized static void init(ProblemDiagnosticsFactory problemDiagnosticsFactory, WarningMode warningMode, BuildOperationProgressEventEmitter buildOperationProgressEventEmitter) {
+        DEPRECATED_FEATURE_HANDLER.init(problemDiagnosticsFactory, warningMode, buildOperationProgressEventEmitter);
```

Comment:
❓ Why did we change to the static problems service accessor here?

Reply:
It's not necessary anymore. All the usage is now in this file by using `ProblemsProgressEventEmitterHolder.get()`

Reply:
Painful question, but just to make sure: can we deprecate stuff from a worker?